### PR TITLE
Tox

### PR DESCRIPTION
**Pull request by BorjaEst**
Create tox configuration with at least the following:
- [x] Common testenv
- [x] Cover testenv
- [x] Pep8 testenv

### CODE REVIEW

**Comment by BorjaEst** [comment]
I cherry picked the basic setup.cfg from #59, but we can merge the final version once it is finished.
For the moment this is only a start with a basic tox configuration (or the idea).

**Comment by BorjaEst** [comment]
In my computer (using conda) tox is having some troubles to install uWSGI.
```sh
    lto1: fatal error: bytecode stream in file ‘/home/borja/miniconda3/envs/flask/lib/python3.8/config-3.8-x86_64-linux-gnu/libpython3.8.a’ generated with LTO version 6.0 instead of the expected 8.1
    compilation terminated.
    lto-wrapper: fatal error: gcc returned 1 exit status
    compilation terminated.
    /home/borja/miniconda3/envs/flask/compiler_compat/ld: error: lto-wrapper failed
    collect2: error: ld returned 1 exit status
    *** error linking uWSGI ***
    ----------------------------------------
```
Probably, uWSGI is not needed to run nor to unit test, so, it might be possible to drag it out of the minimal installation and put them into a "requirements-prod.txt" or similar.

**Comment by BorjaEst** [comment]
At this point it seems some tests are failing:
```
$ tox
...
eosc_perf/tests/facade_test.py ....................................                                  [ 35%]
eosc_perf/tests/io_controller_test.py ......F...F..........F....F...............................     [ 93%]
eosc_perf/tests/json_result_validator_test.py .......                                                [100%]
...
```
For the moment we can leave them (investigate later).
Now I would suggest to continue on the development of tox.ini to add coverage and other features.

**Comment by BorjaEst** [comment]
Tests for pep8 added. An output file is produced "flake8.log" after running `$ tox`. There are some recomendations, but can be fixed in long term.

**Comment by TheChristophe** [comment]
> Probably, uWSGI is not needed to run nor to unit test, so, it might be possible to drag it out of the minimal installation and put them into a "requirements-prod.txt" or similar.

But uwsgi is needed by the application to run in the docker container, which does use setup.cfg, so maybe it should stay in there.

To address your specific issue, I don't know much about conda, but it seems it's using your system's compiler, which is too old. Have you tried it with installing gcc through conda, like here? https://github.com/ContinuumIO/anaconda-issues/issues/6619#issuecomment-351981000

**Comment by BorjaEst** [comment]
Still missing the coverage, but for the rest is working

**Comment by TheChristophe** [comment]
I have removed the _PR_ from the project board; we have the issue on the project board and that links to the PR, and if we merge the PR the issue gets closed too so it's a little redundant to have both.

File: tox.ini [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+[testenv]
+usedevelop = True
+basepython = python3
+whitelist_externals = find
+install_command = pip install -U {opts} {packages}
+# use en_US.UTF-8 as C.UTF-8 doesn't exist in RHEL7
+setenv =
+    VIRTUAL_ENV={envdir}
+    LC_ALL=en_US.UTF-88
```
Comment (TheChristophe) [inline tox.ini:13]: en_US.UTF-8 typo?

Comment (BorjaEst) [reply tox.ini:13]: ooops, yes

**Comment by TheChristophe** [comment]
Do you intend to implement the cover part or should I have a look at it (if you tell me what you mean by it)?

**Comment by BorjaEst** [comment]
I meant cover for [test coverage](https://coverage.readthedocs.io/en/v4.5.x/index.html).
Currently I am on it, trying to call coverage from tox.

Valentin know better tox, but I am trying to undestand how it exaclty works with coverage.

**Comment by BorjaEst** [comment]
To run coverage: `$ tox -e cover`
From myside is ready to merge.

More tox env can be added later.

As main inconvinience, I could not run test in parallel without renaming the tests.... If it is a problem we can roll back I keep searching how to use the regex option on stestr, however, I think this rename will save issues on the future (as it took me 4h to find out where the issue was).

**Comment by TheChristophe** [comment]
I've tested it and it seems to work. It seems to be missing tox in any of the requirements files however, that may be worth adding.
I'll (re-?)rebase this onto master branch because the files changed and commit history seems a bit confused.

**Comment by TheChristophe** [comment]
If I did this correctly, this should be all the current commits. I've also noticed cover is not in the [tox]envlist, so it's not run by default when running `tox`; is that intended?

**Comment by BorjaEst** [comment]
Yes, this is in order to avoid running tests twice.
To run only coverage: $ tox -e cover

**Comment by TheChristophe** [comment]
We may still want to add `tox` to requirements-test.txt as it's currently not present in any of the requirements I think

**Pull request merged**